trin-core/src/types/bus.rs

```diff
@@ -0,0 +1,53 @@
+use anyhow::anyhow;
+use tokio::sync::mpsc;
+
+use crate::jsonrpc::types::{HistoryJsonRpcRequest, StateJsonRpcRequest};
+
+/// Datatype to house all jsonrpc tx channels for available subnetworks
+#[derive(Clone, Debug, Default)]
+pub struct NetworkBus {
+    pub history_jsonrpc_tx: Option<mpsc::UnboundedSender<HistoryJsonRpcRequest>>,
+    pub state_jsonrpc_tx: Option<mpsc::UnboundedSender<StateJsonRpcRequest>>,
+}
+
+impl NetworkBus {
+    pub fn get_history_tx(&self) -> anyhow::Result<mpsc::UnboundedSender<HistoryJsonRpcRequest>> {
+        match self.history_jsonrpc_tx.clone() {
+            Some(val) => Ok(val),
+            None => Err(anyhow!("History subnetwork is not available")),
+        }
+    }
+
+    pub fn set_history_tx(
+        &mut self,
+        tx: mpsc::UnboundedSender<HistoryJsonRpcRequest>,
+    ) -> anyhow::Result<()> {
+        match self.history_jsonrpc_tx.clone() {
+            Some(_) => Err(anyhow!("History tx has already been set")),
+            None => {
+                self.history_jsonrpc_tx = Some(tx);
+                Ok(())
+            }
+        }
```

Comment:
The idea is that once the `history_jsonrpc_tx` channel has been set, it should never need to be set again. If we try to set it again, once it's already been set, then that reveals a flaw in our setup. Not sure if this is a valid concern or not? Wdyt?

Reply:
I think here we mix two responsibilities: 1. "Make sure the field is not set" and 2. "Set the field if it is not." So maybe a different function name is more suitable, something like `try_set_history_tx`?

Reply:
To satisfy the pattern of "set at most once" we should pass that value optionally in the constructor and only provider an accessor method. The current pattern suggests a flaw in the design.

Reply:
> we should pass that value optionally in the constructor and only provider an accessor method.

This doesn't seem possible given our current setup. When initializing trin, we essentially initialize this "bus" (via the `HeaderOracle`) and pass it into every subnetwork's initialization - during which that subnetwork attaches its respective jsonrpc tx. It seems to me it would require a major re-architecting to avoid this, unless I'm missing what you're suggesting?